### PR TITLE
[runtime] Fix memory leaks related to short-lived strings

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

**Comment by kumpera** [comment]
LGTM

File: mono/mini/aot-runtime.c [hunk 1]
```diff
@@ -1986,7 +1986,10 @@ load_aot_module (MonoAssembly *assembly, gpointer user_data)
 				mono_trace (G_LOG_LEVEL_INFO, MONO_TRACE_AOT, "AOT module '%s' not found: %s\n", aot_name, err);
 				g_free (err);
 
-				aot_name = g_strdup_printf ("%s/mono/aot-cache/%s/%s%s", mono_assembly_getrootdir(), MONO_ARCHITECTURE, g_path_get_basename (assembly->image->name), MONO_SOLIB_EXT);
+				g_free (aot_name);
+				char *basename = g_path_get_basename (assembly->image->name);
```
Comment (tritao) [inline mono/mini/aot-runtime.c]: Can we use C99 variables declared in middle of block now?

Comment (alexrp) [inline mono/mini/aot-runtime.c]: Yep, we've been C99 since around Feb 2015.

**Comment by lewurm** [comment]
@monojenkins merge

**Pull request merged**